src/services/clinical/index.js

```diff
@@ -137,58 +139,80 @@ const getClinicalSubmissionData = async (programShortName, Authorization) => {
 
 const getClinicalData = async (variables, Authorization) => {
 	const { programShortName, filters } = variables;
-
-	const query = new URLSearchParams(filters).toString();
+	const { completionState, entityTypes, donorIds, submitterDonorIds } = filters;
+	const queryString = buildClinicalInputQueryString(filters);
 
 	const url = urlJoin(
 		CLINICAL_SERVICE_ROOT,
 		`/clinical/program/`,
 		programShortName,
-		`/clinical-data?${query}`,
+		`/clinical-data?${queryString}`,
 	);
 
 	const response = await fetch(url, {
-		method: 'get',
+		method: 'post',
 		headers: { Authorization },
+		body: {
+			completionState,
+			entityTypes,
+			donorIds: donorIds?.map((id) => Number(id)) || [],
```

Comment:
This line I would review b/c there's no handling for where `Number(id) = NaN`
I would want to make sure front end / clinical validation and filtering are in place

Reply:
Good point. I think in this case nothing will break since it will just not find any donors that match ID=NaN. I'll add a filter.

Reply:
Hmm good question but I'm not sure if we should throw out the whole request for one bad id

For the Search Bar and Filter Modal, if you enter one valid DonorId and one invalid ID, you still receive the results for the correct ID, and the invalid IDs are just ignored. These use a different query (the search query) but I think the same principle should probably apply, especially if a user submits this kind of search

Reply:
Error messages for invalid IDs in the clinical response and on the UI could point out bad user input

Reply:
So the GQL Schema will enforce the filter be an array of numbers, making my conversion redundant... I added in a filter for NaN's so we only send valid numeric IDs.